api/api.go

```diff
@@ -339,12 +339,16 @@ func encodeBody(obj interface{}) (io.Reader, error) {
 // requireOK is used to wrap doRequest and check for a 200
 func requireOK(d time.Duration, resp *http.Response, e error) (time.Duration, *http.Response, error) {
 	if e != nil {
-		return d, resp, e
+		if resp != nil {
```

Comment:
If resp comes from a http.Client.Do, there's no need to do this but I am trying not to take any assumptions.
Even if it comes from http.Client.Do, there should be no harm in closing it here.